samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsManager.java

```diff
@@ -80,20 +82,20 @@
   private final SystemStream diagnosticSystemStream;
 
   public DiagnosticsManager(String jobName, String jobId, Map<String, ContainerModel> containerModels,
-      Integer containerMemoryMb, Integer containerNumCores, Integer numStoresWithChangelog, String containerId,
-      String executionEnvContainerId, String taskClassVersion, String samzaVersion, String hostname,
+      Integer containerMemoryMb, Integer containerNumCores, Integer numStoresWithChangelog, Long maxHeapSizeBytes, Integer containerThreadPoolSize,
```

Comment:
Are these arguments nullable? If not, can you please use the primitive versions?